### ISSUE TITLE
Force strict

### ISSUE THREAD

**Issue by fatcerberus** [svaarala/duktape#713]

**Comment by saghul** [comment]
Yeah, this would be great. I see the compile functions ([duk_compile](http://duktape.org/api.html#duk_compile) which get a `flags` argument allow for this, but not the eval functions.

**Comment by fatcerberus** [comment]
@saghul Thanks for the tip on `duk_compile()` - somehow I missed that. :-)

**Comment by svaarala** [comment]
@fatcerberus Yes, `duk_compile()` is the main actual compilation function which handles also eval code and has a force strict flag. The convenience API calls like `duk_eval()` are more restricted in their usage, but you can just use `duk_compile()` directly.

API-wise I'm planning on removing some (or maybe even all) of the eval/compile variants. What would be left is one function for compiling functions and another for eval'ing an input, with each getting a flags field. The situation would be closed to `duk_def_prop()` in that sense. There are usually quite few eval/compile call sites so the convenience variants don't necessarily defend their existence very well.
